examples/linear_model.ipynb

```diff
@@ -0,0 +1,306 @@
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For our `threshold_met()`, we have defined the function to return a boolean function when our thrown object has met the initial state of 'impact', or in other words, has hit the ground. This occurs when our height is less than or equal to 0, and our speed is less than 0!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class ThrownObject(ThrownObject):\n",
+    "    def threshold_met(self, x):\n",
+    "        return {\n",
+    "            'falling': x['v'] < 0,\n",
+    "            'impact': x['x'] <= 0\n",
+    "        }"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, for our `event_state()`, we have defined our events to be when our object has hit the ground, or has made 'impact'. The `event_state()` function specifically returns the current state of our object at any given point of it's journey through the air!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class ThrownObject(ThrownObject):\n",
+    "    def event_state(self, x): \n",
+    "        x_max = x['x'] + np.square(x['v'])/(-self.parameters['g']*2)\n",
+    "        return {\n",
+    "            'falling': np.maximum(x['v']/self.parameters['throwing_speed'],0),\n",
+    "            'impact': np.maximum(x['x']/x_max,0) if x['v'] < 0 else 1\n",
+    "        }"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "With these functions created, we can now run our ThrownObject Model!\n",
+    "\n",
+    "In this example, we will initialize our ThrownObject as `m`, and we'll define our `future_loading()` function (more information about future loading: https://nasa.github.io/progpy/prog_models_guide.html#future-loading) and our `simulate_to_threshold()` function (more information about simulation: https://nasa.github.io/progpy/prog_models_guide.html#simulation) to demonstrate the movement of the thrown object in air!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m = ThrownObject()\n",
+    "def future_loading(t, x=None):\n",
+    "    return m.InputContainer({})\n",
+    "save = m.simulate_to_threshold(future_loading, print = True, save_freq=1, threshold_keys='impact', dt=0.1)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We'll also demonstrate how this looks plotted on a graph."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "save.outputs.plot(title='generated model')\n",
+    "plt.show()"
+   ]
+  }
```

Comment:
Let's add a markdown block at the end where we talk about what we see here. How the position follows a parabola as expected. 

And perhaps a final conclusion block that brings it all together. Mentioning how LinearModel provides a format for building simple models. And how, if a model can be defined as a linear model, it opens up the opportunity to use additional algorithms in prog_algs, that require a LinearModel